### DISCUSSION TITLE
Does the React compiler do tree shaking ?

### DISCUSSION THREAD

**Discussion by KingAmo** [reactwg/react-compiler#49] [General Questions] [answered]
Does React compiler do tree shaking  or it will be able to do tree shaking in the future?

**Comment by slorber** [comment]
Afaik it doesn't do that and doesn't plan to

The React Compiler only operates on files in isolation, it doesn't see or modify your app's module graph. It's currently exposed as a Babel plugin, and Babel plugins do not have this ability.

Tree shaking is a bundler feature, and React Compiler is not a bundler.

**Reply by KingAmo** [reply]
it seems it do dead code elimination, see https://github.com/facebook/react/blob/main/compiler/docs/DESIGN_GOALS.md

**Reply by poteto** [reply, marked as answer]
There are no current plans to add tree shaking to the compiler.
